### DISCUSSION TITLE
Fail on Win10: create socket; Fail on Win7:  GetHostNameW not found

### DISCUSSION THREAD

**Discussion by FrankFOS** [localsend/localsend#2041] [Discussion (English, International)]
Trying to migrate to Win11 machines:
Win 10 Pro 22H2 32GB.  LocalSend-1.16.1-windows-x86-64.exe:
Can send to a Win11 machine, but cannot receive-- trying to start server (as user or admin) get:
"SocketException: Failed to create server socket (OS Error: an attempt was made to access a socket in a way forbidden 
by its access permissions, errno = 10013), address = 0.0.0.0 port = 53317"

Win7 Home Premium SP1 12GB. LocalSend-1.16.1-windows-x86-64.exe:
Can not even start LocalSend-- "The procedure entry point GetHostNameW could not be located in the DLL WS2_32.dll"

Things seem OK on Win11.

**Comment by glockmane** [comment]
I'm having the same Windows 7 Problem (sadly I cannot upgrade two Win 7 Machines because of software compatiblity..).. Is this fixable? Already thought about replacing the DLL with the one from Windows 10/11..

**Comment by TheAndrey** [comment]
This is a latest version that supports **Windows 7**
https://github.com/localsend/localsend/releases/tag/v1.12.0

**Reply by glockmane** [reply]
Thanks! Would still be interesting if it is intended that Windows 7 support is not working anymore cause a few changelogs earlier it was mentioned that it was restored..

**Reply by tony9527167** [reply]
> This is a latest version that supports **Windows 7** https://github.com/localsend/localsend/releases/tag/v1.12.0

I confirmed that [v1.15.4](https://github.com/localsend/localsend/releases/tag/v1.15.4) also works on Windows 7 64-bit :)

**Comment by I21b** [comment]
#2618
